app/src/main/java/ayds/lisboa/songinfo/home/view/SongDescriptionHelperImpl.kt

```diff
@@ -1,25 +1,31 @@
 package ayds.lisboa.songinfo.home.view
 
-import ayds.lisboa.songinfo.home.model.entities.Song.EmptySong
 import ayds.lisboa.songinfo.home.model.entities.Song
+import ayds.lisboa.songinfo.home.model.entities.Song.EmptySong
 import ayds.lisboa.songinfo.home.model.entities.Song.SpotifySong
 
 interface SongDescriptionHelper {
     fun getSongDescriptionText(song: Song = EmptySong): String
 }
 
-internal class SongDescriptionHelperImpl : SongDescriptionHelper {
+internal class SongDescriptionHelperImpl(private val releaseDateFactory: ReleaseDateFactory): SongDescriptionHelper {
+
     override fun getSongDescriptionText(song: Song): String {
         return when (song) {
-            is SpotifySong ->
-                "${
-                    "Song: ${song.songName} " +
-                            if (song.isLocallyStored) "[*]" else ""
-                }\n" +
-                        "Artist: ${song.artistName}\n" +
-                        "Album: ${song.albumName}\n" +
-                        "Year: ${song.year}"
+            is SpotifySong -> buildSpotifySongDescription(song)
             else -> "Song not found"
         }
     }
-}
\ No newline at end of file
+
+    private fun buildSpotifySongDescription(song: SpotifySong): String {
+        return "${
+            "Song: ${song.songName} " +
+                    if (song.isLocallyStored) "[*]" else ""
+        }\n" +
+                "Artist: ${song.artistName}\n" +
+                "Album: ${song.albumName}\n" +
+                "Release date: ${createReleaseDate(song)}"
+    }
+
+    private fun createReleaseDate(song: SpotifySong) = releaseDateFactory.get(song).formatDate()
```

Comment:
Para mantener el nivel de abstracción pueden utilizar una función de extension:
```suggestion
    private fun SpotifySong.createReleaseDate() = releaseDateFactory.get(this).formatDate()
```